server_src/lib/dataFunctions/fileClean.js

```diff
@@ -68,107 +68,113 @@ class FileClean {
-            );
-            sortedFileList.push(sortedFile);
-            // console.log(sortedFileList);
-          }
-        } catch (e) {
-          console.log(e);
-        }
-      }
-      const newFileList = {
-        fileList: sortedFileList,
-        filecount: 0,
-        folderList: [],
-        folderCount: 0,
-      };
-      newFileList.filecount = fileList.fileCount;
-      newFileList.folderList = fileList.folders;
-      newFileList.folderCount = fileList.folderCount;
-      cleanFileList[farmPrinter.sortIndex] = newFileList;
-      if (typeof farmPrinter.systemChecks !== "undefined") {
-        farmPrinter.systemChecks.cleaning.file.status = "success";
-        farmPrinter.systemChecks.cleaning.file.date = new Date();
+    const { fileList, sortIndex } = farmPrinter;
+
+    if (!fileList) {
+      logger.error("File Cleaner failed: farmPrinter:fileList not defined.");
+      return;
+    }
+
+    // NaN, object, undefined here
+    if (Number.isNaN(sortIndex) || isNaN(sortIndex)) {
+      logger.error(`File Cleaner failed: farmPrinter:sortIndex is NaN (${sortIndex})`);
+      return;
+    }
+
+    // null, string caught here
+    if (!sortIndex && !Number.isInteger(sortIndex)) {
+      logger.error(`File Cleaner failed: farmPrinter:sortIndex not defined (${sortIndex})`);
+      return;
+    }
+
+    if (sortIndex < 0) {
+      logger.error(`File Cleaner failed: farmPrinter:sortIndex cannot be negative (${sortIndex})`);
+      return;
+    }
+
+    if (!!farmPrinter.systemChecks) {
+      farmPrinter.systemChecks.cleaning.file.status = "warning";
+    }
+    const printCost = farmPrinter.costSettings;
+    const sortedFileList = [];
+    if (!!fileList?.files) {
+      for (let file of fileList.files) {
+        const sortedFile = {
+          path: file.path,
+          fullPath: file.fullPath,
+          display: file.display,
+          name: file.name,
+          uploadDate: file.date,
+          fileSize: file.size,
+          thumbnail: file.thumbnail,
+          toolUnits: "",
+          toolCosts: "",
+          success: file.success,
+          failed: file.failed,
+          last: file.last,
+          expectedPrintTime: file.time,
+          printCost: getPrintCostNumeric(file.time, printCost),
+        };
+        sortedFile.toolUnits = FileClean.getUnits(
+          selectedFilament,
+          file.length,
+        );
+        sortedFile.toolCosts = FileClean.getCost(
+          selectedFilament,
+          sortedFile.toolUnits,
+        );
+        sortedFileList.push(sortedFile);
       }
-      logger.info("File Information cleaned and ready for consumption...");
-      return true;
-    } catch (e) {
-      console.log("MASSIVE FILE ERROR!!!!", e);
     }
+
+    // TODO oof not the sortIndex again... this is SUPER BUG PRONE
+    // sortIndex should be a supporting math tool
+    cleanFileList[sortIndex] = {
+      fileList: sortedFileList,
+      filecount: fileList.fileCount || 0,
```

Comment:
`filecount` is not camel case, this is more bug prone (for me) than you might think

Reply:
Aye was a goof ages ago I never changed.